app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala

```diff
@@ -38,6 +39,17 @@ case class WalletRoutes(wallet: UnlockedWalletApi, node: Node)(
     Server.httpSuccess(sendString)
   }
 
+  private def handleBroadcastable(
+      tx: Transaction,
+      noBroadcast: Boolean): String = {
+    if (noBroadcast) {
+      tx.hex
+    } else {
+      node.broadcastTransaction(tx)
+      tx.txIdBE.hex
```

Comment:
I think it is problematic that we don't propogate error messages from `node.broadcastTransaction` to the returned value from `handleBroadcastable`. It should return a `Future[String]` and return the txid inside of a map on the broadcast

Reply:
And then we need to make sure that the error gets propagated well to the cli (as opposed to just like a 5** error message), this can be tested by temporarily replacing the broadcast with a `Future.failed`

Reply:
I agree with this, also why `String`? Shouldn't it at least be a hash?

Reply:
> I agree with this, also why String? Shouldn't it at least be a hash? Types are our friends :-)

It is either a hash or a `Transaction`